File: leetcode3/황은지/648. Replace Words.js
/**
 * @param {string[]} dictionary
 * @param {string} sentence
 * @return {string}
 */
var replaceWords = function (dictionary, sentence) {
  dictionary.sort((a, b) => a.length - b.length);
  const splited = sentence.split(" ");

  for (let i = 0; i < splited.length; i++) {
    const word = splited[i];
    for (let dicWord of dictionary) {
      if (word.startsWith(dicWord)) {
        splited[i] = dicWord;
        break;
      }
    }
  }

  return splited.join(" ");
};

File: leetcode3/김민제/648_Replace_Words.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        dictionary.sort(key=len)
        new_sentence = ''
        arr = sentence.split(" ")
        for s in arr:
            replaced=False
            for word in dictionary:
                if s.startswith(word):
                    new_sentence += word
                    replaced=True
                    break

            if not replaced:
                new_sentence += s
            new_sentence += ' '

        return new_sentence.rstrip()
